armory/scenarios/image_classification.py

```diff
@@ -26,6 +27,12 @@ def _evaluate(self, config: dict) -> dict:
 
         model_config = config["model"]
         classifier, preprocessing_fn = load_model(model_config)
+        defense = config.get("defense")
+        if defense:
+            defense_type = defense["type"]
+            if defense_type in ["Preprocessor", "Postprocessor"]:
```

Comment:
to prevent nested double "if" statements everywhere you use this, you could do:
```
defense = config.get("defense", {})
defense_type = defense.get("type")
if defense_type in ["Preprocessor", "Postprocessor"]:
    ...
...
if defense_type in ["Trainer", "Transformer"]:
    ...
```